### PR TITLE
feat: List extension: fast remove method by swap elements

### PR DESCRIPTION
**Pull request by Yurii-Tor**
## Purpose of this PR
Fast method for removing elements from List implemented.

This feature is often on demand. It changes an order of elements inside the list, so not recomended if order matters. Instead of that LinkedList may be used if it feats needs (it's not an IList type)

## Testing status

 - No tests have been added.

### Manual testing status
Tested in Unity Editor. See link below:
https://share.getcloudapp.com/RBulD2kG

## Comments to reviewers
Notice that 'Remove' method deletes only one element, so the predicate evaluation deletes only the first founded. If you need to delete all evaluated elements, use RemoveAll method, which is also fast and doesn't allocate additional resources.

### CODE REVIEW

**Review by stan-osipov — CHANGES_REQUESTED** [review]
Looks good.
But please convert this test:

> Tested in Unity Editor. See link below:
> https://share.getcloudapp.com/RBulD2kG

To the actual Uini test.

**Comment by stan-osipov** [comment]
@Yurii-Tor any luck with the test?

**Comment by Yurii-Tor** [comment]
> @Yurii-Tor any luck with the test?

I'm working on it now. Just started

**Review by stan-osipov — CHANGES_REQUESTED** [review]
Super minor naming requets.
Thank you for the tests.

File: Tests/Editor/Extensions/CSharpTests.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using StansAssets.Foundation.Extensions;
+using UnityEngine;
+
+namespace StansAssets.Foundation.Patterns.EditorTests 
+{
+    public class CSharpTests {
```
[the remark this answers is not in the document]
Comment (Yurii-Tor) [reply Tests/Editor/Extensions/CSharpTests.cs:10]: Ok. No problem. I was confused by the file structure and names.
Thanks for reply! =)

**Review by stan-osipov — APPROVED** [review]
lovely!

**Pull request merged**